### PR TITLE
Change Tasks to use Kubernetes Pods instead of Jobs (#41)

### PR DESCRIPTION
**Pull request by abhinavdas**
This change is to axmon to generate a Pod spec for Tasks instead of
generating a Job spec. It will now be up to workflow executor and
workflow ADC to ensure that tasks that did not complete due to node
failure or other conditions are restarted.

### CODE REVIEW

File: .argo/test-yamls/pod-throughput-test.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ name: quick-exit
 description: does nothing and exits quickly
 resources:
   mem_mib: 32
-  cpu_cores: 0.1
+  cpu_cores: 0.01
```
Comment (lcjlcj) [inline .argo/test-yamls/pod-throughput-test.yaml]: Is this for test only? Such low cpu request could potentially trigger kernel issues.

File: platform/source/lib/ax/platform/task.py [hunk 2]
```diff
@@ -101,275 +96,134 @@ def create(self, conf):
...
+                        break
+            except Exception:
+                pass
+
+        return ret_status
+
+    def delete(self, force=False):
         """
         Delete the task from kubernetes and returns the final status
-        Args:
```
Comment (lcjlcj) [inline platform/source/lib/ax/platform/task.py]: Nice to delete this.

File: .argo/test-yamls/pod-throughput-test.yaml (hunk 1 above)
Comment (abhinavdas) [reply .argo/test-yamls/pod-throughput-test.yaml]: This is for test only. We still have num pods <= 80 per node and we do not set cpu limit

File: platform/source/lib/ax/platform/task.py [hunk 3]
```diff
@@ -101,275 +96,134 @@ def create(self, conf):
...
+        #     // with a container exit code of 0, and the system is not going to restart any of these containers.
+        #     PodSucceeded PodPhase = "Succeeded"
+        #     // PodFailed means that all containers in the pod have terminated, and at least one container has
+        #     // terminated in a failure (exited with a non-zero exit code or was stopped by the system).
+        #     PodFailed PodPhase = "Failed"
+        #     // PodUnknown means that for some reason the state of the pod could not be obtained, typically due
+        #     // to an error in communicating with the host of the pod.
+        #     PodUnknown PodPhase = "Unknown"
+        # )
+        pending = True if status.status.phase == "Pending" else False
```
Comment (shrinandj) [inline platform/source/lib/ax/platform/task.py]: Does this have to be case-insensitive?

File: platform/source/lib/ax/platform/task.py [hunk 4]
```diff
@@ -101,275 +96,134 @@ def create(self, conf):
...
         }
-        if active:
-            reason = self.get_last_pod_waiting_reason()
-            if reason:
-                status['reason'] = reason
 
-            # do some cleanup for pods that have failed due to system errors such as mismatch
-            # betweeen kubernetes scheduler and kubelet resources
+        if pending:
+            # If the pod is pending it is likely stuck on container image pull failure which
```
Comment (shrinandj) [inline platform/source/lib/ax/platform/task.py]: This comment looks incomplete..

**Pull request merged**